src/main/drivers/serial_uart_hal.c

```diff
@@ -199,13 +209,24 @@ void uartReconfigure(uartPort_t *uartPort)
 void uartTryStartTxDMA(uartPort_t *s)
 {
     ATOMIC_BLOCK(NVIC_PRIO_SERIALUART_TXDMA) {
-        if (IS_DMA_ENABLED(s->txDMAResource)) {
+#if 0
```

Comment:
Disabling this is almost certainly not what we want - re-enable or remove.

Reply:
Oops...

Reply:
Okay, this section was disabled as G4 does not automatically disable the channel (by hardware resetting channel EN bit); the channel remained enabled and the test always succeeded. Now the EN bit is reset in `uartDmaIrqHandler()`, so the section is enabled and the check is valid.